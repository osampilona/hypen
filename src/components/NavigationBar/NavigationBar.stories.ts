import type { Meta, StoryObj } from "@storybook/react";
import NavigationBar from "@/components/NavigationBar/NavigationBar";

interface INavigationBarProps {
  labelPartner?: string;
  isSimpleNavbar?: boolean;
}

const meta: Meta<typeof NavigationBar> = {
  title: "Atomic Components/Navigation Bar",
  component: NavigationBar,
  parameters: {},
  tags: ["autodocs"],
  argTypes: {
    labelPartner: { control: "text" },
    isSimpleNavbar: { control: "boolean" },
  },
};

export default meta;

type Story = StoryObj<INavigationBarProps>;

export const Default: Story = {
  args: {
    labelPartner: "Become a Partner",
    isSimpleNavbar: false,
  },
};
