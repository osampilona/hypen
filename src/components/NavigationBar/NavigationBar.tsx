"use client";
import React, { useState, useEffect } from "react";
import navigationBar from "@/components/NavigationBar/navigationBar.module.scss";
import SearchBar from "../SearchBar/SearchBar";
import { HiOutlineUserCircle } from "react-icons/hi2";
import BigScreenLogoIcon from "@/assets/icons/BigScreenLogoIcon";
import SubNavigationBar from "../SubNavigationBar/SubNavigationBar";
import { serviceItems } from "@/data/serviceList";

export interface INavigationBarProps {
  labelPartner: string;
}

const NavigationBar = (props: INavigationBarProps) => {
  const [viewportWidth, setViewportWidth] = useState<number | undefined>(
    undefined,
  );

  useEffect(() => {
    const handleResize = () => {
      setViewportWidth(window.innerWidth);
    };
    handleResize();
    window.addEventListener("resize", handleResize);

    return () => window.removeEventListener("resize", handleResize);
  }, []);

  return (
    <div className={navigationBar.main}>
      <div className={navigationBar.container}>
        <div className={navigationBar.screen}>
          {(viewportWidth ?? 0) <= 1024 ? null : (
            <div className={navigationBar.icons}>
              <BigScreenLogoIcon />
              <div className={navigationBar.label}>
                <p>{props.labelPartner}</p>
                <HiOutlineUserCircle size={36} />
              </div>
            </div>
          )}
        </div>
        <SearchBar labelWhat={"What"} labelWhere={"Where"} labelWhen={"When"} />
      </div>
      <SubNavigationBar items={serviceItems} />
    </div>
  );
};

export default NavigationBar;
