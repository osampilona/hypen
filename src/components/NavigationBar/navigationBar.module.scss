/* navigationBar.module.scss */
.container {
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  color: #686868;
  position: relative;
  padding: 1rem;
  position: fixed;
  z-index: 100;
  height: 5.3125rem;
  background-color: #fbf7f5;
  top: 0;
  left: 0;

  .screen {
    display: flex;
    width: -webkit-fill-available;

    .icons {
      display: flex;
      align-items: center;
      justify-content: space-between;
      width: 100%;

      padding: 1rem 0;

      .label {
        gap: 0.5rem;
        display: flex;
        align-items: center;
      }
    }
  }

  @media (min-width: 1025px) {
    flex-direction: row;
  }
}
