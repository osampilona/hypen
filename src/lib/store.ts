import { configureStore } from "@reduxjs/toolkit";
import themeReducer from "@/lib/features/theme/theme";
import filterReducer from "@/lib/features/filters/filters";

export const store = configureStore({
  reducer: {
    theme: themeReducer,
    filters: filterReducer,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export default store;
