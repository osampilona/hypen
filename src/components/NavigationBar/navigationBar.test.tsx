import { describe, test, expect } from "vitest";
import { render, screen, fireEvent } from "@testing-library/react";
import { Provider } from "react-redux";
import store from "@/lib/store"; // Use your actual store configuration
import NavigationBar from "@/components/NavigationBar/NavigationBar";

describe("NavigationBar", () => {
  test("renders correctly", () => {
    const { container } = render(
      <Provider store={store}>
        <NavigationBar labelPartner="Become a partner" />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });

  test("renders the labelPartner correctly", () => {
    render(
      <Provider store={store}>
        <NavigationBar labelPartner="Become a partner" />
      </Provider>,
    );
    const labelPartners = screen.getAllByText("Become a partner");
    expect(labelPartners.length).toBeGreaterThan(0);
  });

  test("renders user icon", () => {
    render(
      <Provider store={store}>
        <NavigationBar labelPartner="Become a partner" />
      </Provider>,
    );
    const userIcon = screen.queryAllByLabelText("profile-icon");
    expect(userIcon).toBeTruthy();
  });

  test("toggles theme correctly", () => {
    render(
      <Provider store={store}>
        <NavigationBar labelPartner="Become a partner" />
      </Provider>,
    );

    // Initial theme should be light
    expect(document.body.classList.contains("dark-theme")).toBe(false);

    // Click the theme toggle button
    const themeToggleButton = screen.getAllByRole("button")[0]; // Adjust selector if needed
    fireEvent.click(themeToggleButton);

    // After clicking, theme should be dark
    expect(document.body.classList.contains("dark-theme")).toBe(true);

    // Click the theme toggle button again
    fireEvent.click(themeToggleButton);

    // After clicking again, theme should be light
    expect(document.body.classList.contains("dark-theme")).toBe(false);
  });
});
