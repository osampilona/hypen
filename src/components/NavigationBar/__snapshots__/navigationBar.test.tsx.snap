// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavigationBar > renders correctly 1`] = `
<div>
  <div
    class="_main_3d2020"
  >
    <div
      class="_container_3d2020"
    >
      <div
        class="_screen_3d2020"
      >
        <div
          class="_icons_3d2020"
        >
          <svg
            fill="none"
            height="48"
            viewBox="0 0 72 48"
            width="72"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="36"
              cy="24"
              fill="#FAFAFA"
              r="23.5"
              stroke="#282828"
            />
            <rect
              fill="#FAFAFA"
              height="10.5763"
              width="70.9859"
              y="18.6272"
            />
            <path
              d="M0.948153 31V20.0909H3.25462V24.592H7.93679V20.0909H10.2379V31H7.93679V26.4936H3.25462V31H0.948153ZM13.0004 20.0909H15.5838L18.0714 24.7891H18.1779L20.6655 20.0909H23.2489L19.2699 27.1435V31H16.9794V27.1435L13.0004 20.0909ZM26.0087 31V20.0909H30.3127C31.1401 20.0909 31.845 20.2489 32.4274 20.565C33.0098 20.8775 33.4537 21.3125 33.7591 21.87C34.068 22.424 34.2225 23.0632 34.2225 23.7876C34.2225 24.5121 34.0662 25.1513 33.7537 25.7053C33.4412 26.2592 32.9885 26.6907 32.3954 26.9996C31.8059 27.3086 31.0922 27.4631 30.2541 27.4631H27.5108V25.6147H29.8812C30.3251 25.6147 30.6909 25.5384 30.9785 25.3857C31.2697 25.2294 31.4863 25.0146 31.6284 24.7411C31.774 24.4641 31.8468 24.1463 31.8468 23.7876C31.8468 23.4254 31.774 23.1094 31.6284 22.8395C31.4863 22.5661 31.2697 22.3548 30.9785 22.2056C30.6873 22.0529 30.318 21.9766 29.8706 21.9766H28.3152V31H26.0087ZM37.2206 31V20.0909H39.5271V24.592H44.2093V20.0909H46.5104V31H44.2093V26.4936H39.5271V31H37.2206ZM49.912 31V20.0909H57.2629V21.9925H52.2185V24.592H56.8847V26.4936H52.2185V29.0984H57.2842V31H49.912ZM69.7212 20.0909V31H67.729L62.983 24.1339H62.9031V31H60.5966V20.0909H62.6207L67.3295 26.9517H67.4254V20.0909H69.7212Z"
              fill="#282828"
            />
          </svg>
          <div
            class="_label_3d2020"
          >
            <p>
              Become a partner
            </p>
            <svg
              aria-hidden="true"
              data-testid="user-icon"
              fill="none"
              height="36"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              width="36"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17.982 18.725A7.488 7.488 0 0 0 12 15.75a7.488 7.488 0 0 0-5.982 2.975m11.963 0a9 9 0 1 0-11.963 0m11.963 0A8.966 8.966 0 0 1 12 21a8.966 8.966 0 0 1-5.982-2.275M15 9.75a3 3 0 1 1-6 0 3 3 0 0 1 6 0Z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="_container_528985"
      >
        <div
          class="_search_528985"
        >
          <div
            class="_icon_528985"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10.25 2a8.25 8.25 0 0 1 6.34 13.53l5.69 5.69a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215l-5.69-5.69A8.25 8.25 0 1 1 10.25 2ZM3.5 10.25a6.75 6.75 0 1 0 13.5 0 6.75 6.75 0 0 0-13.5 0Z"
              />
            </svg>
            <p
              class="_label_528985 "
              data-testid=""
            >
              What
            </p>
          </div>
          <div
            class="_separator_528985"
          />
          <div
            class="_icon_528985"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 48c-79.5 0-144 61.39-144 137 0 87 96 224.87 131.25 272.49a15.77 15.77 0 0 0 25.5 0C304 409.89 400 272.07 400 185c0-75.61-64.5-137-144-137z"
                fill="none"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="32"
              />
              <circle
                cx="256"
                cy="192"
                fill="none"
                r="48"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="32"
              />
            </svg>
            <p
              class="_label_528985 "
              data-testid=""
            >
              Where
            </p>
          </div>
          <div
            class="_separator_528985"
          />
          <div
            class="_icon_528985"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.75 0a.75.75 0 0 1 .75.75V3h9V.75a.75.75 0 0 1 1.5 0V3h2.75c.966 0 1.75.784 1.75 1.75v16a1.75 1.75 0 0 1-1.75 1.75H3.25a1.75 1.75 0 0 1-1.75-1.75v-16C1.5 3.784 2.284 3 3.25 3H6V.75A.75.75 0 0 1 6.75 0ZM21 9.5H3v11.25c0 .138.112.25.25.25h17.5a.25.25 0 0 0 .25-.25Zm-17.75-5a.25.25 0 0 0-.25.25V8h18V4.75a.25.25 0 0 0-.25-.25Z"
              />
            </svg>
            <p
              class="_label_528985 "
              data-testid=""
            >
              When
            </p>
          </div>
        </div>
        <div
          class="_icon_528985"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18.75 12.75h1.5a.75.75 0 0 0 0-1.5h-1.5a.75.75 0 0 0 0 1.5ZM12 6a.75.75 0 0 1 .75-.75h7.5a.75.75 0 0 1 0 1.5h-7.5A.75.75 0 0 1 12 6ZM12 18a.75.75 0 0 1 .75-.75h7.5a.75.75 0 0 1 0 1.5h-7.5A.75.75 0 0 1 12 18ZM3.75 6.75h1.5a.75.75 0 1 0 0-1.5h-1.5a.75.75 0 0 0 0 1.5ZM5.25 18.75h-1.5a.75.75 0 0 1 0-1.5h1.5a.75.75 0 0 1 0 1.5ZM3 12a.75.75 0 0 1 .75-.75h7.5a.75.75 0 0 1 0 1.5h-7.5A.75.75 0 0 1 3 12ZM9 3.75a2.25 2.25 0 1 0 0 4.5 2.25 2.25 0 0 0 0-4.5ZM12.75 12a2.25 2.25 0 1 1 4.5 0 2.25 2.25 0 0 1-4.5 0ZM9 15.75a2.25 2.25 0 1 0 0 4.5 2.25 2.25 0 0 0 0-4.5Z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="_container_ec278e"
      data-testid="subnavigation-bar"
    >
      <div
        class="_links_ec278e"
      >
        <div>
          <a
            href="/#"
          >
            <svg
              data-testid="subnavigation-icon-0"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 15 15"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M0.949988 4.48455C0.949988 5.34059 1.64395 6.03455 2.49999 6.03455C3.35603 6.03455 4.04999 5.34059 4.04999 4.48455C4.04999 3.6285 3.35603 2.93455 2.49999 2.93455C1.64395 2.93455 0.949988 3.6285 0.949988 4.48455ZM2.49999 6.93455C1.14689 6.93455 0.0499878 5.83764 0.0499878 4.48455C0.0499878 3.13145 1.14689 2.03455 2.49999 2.03455C3.85309 2.03455 4.94999 3.13145 4.94999 4.48455C4.94999 4.80813 4.88726 5.11707 4.77329 5.39985L5.73655 6.04413C5.72943 6.06656 5.72269 6.08914 5.71635 6.11188L5.56288 6.66186L5.21005 6.89773L4.21554 6.23366C3.77357 6.6672 3.168 6.93455 2.49999 6.93455ZM0.949991 10.5C0.949991 9.64394 1.64395 8.94998 2.49999 8.94998C3.35603 8.94998 4.04999 9.64394 4.04999 10.5C4.04999 11.356 3.35603 12.05 2.49999 12.05C1.64395 12.05 0.949991 11.356 0.949991 10.5ZM2.49999 8.04998C1.14689 8.04998 0.0499908 9.14688 0.0499908 10.5C0.0499908 11.8531 1.14689 12.95 2.49999 12.95C3.85309 12.95 4.94999 11.8531 4.94999 10.5C4.94999 10.1803 4.88877 9.87497 4.77743 9.595L15 2.75737L14.0486 2.85737C12.7077 2.9983 11.4091 3.40895 10.231 4.06461L7.15646 5.77564C6.92307 5.90553 6.75134 6.12339 6.67955 6.38066L6.42653 7.28737L4.22437 8.75957C3.78166 8.32091 3.17246 8.04998 2.49999 8.04998ZM7.14435 9.2149L7.15646 9.22176L10.231 10.9328C11.4091 11.5884 12.7077 11.9991 14.0486 12.14L15 12.24L8.81072 8.1003L7.14435 9.2149Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
            Hair
          </a>
        </div>
        <div>
          <a
            href="/#"
          >
            <svg
              data-testid="subnavigation-icon-1"
              fill="currentColor"
              height="24"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M211.3 80.89C122.5 81.18 56.21 109.7 18.89 145l12.36 13C93.43 99.36 258.4 54.4 485.6 176.8l8.6-15.8c-108.8-58.6-204.6-80.37-282.9-80.11zm-96.9 84.81c-55.98 21.9-81.16 65.6-96.64 94.4l15.86 8.6c15.24-28.5 37.13-66.6 87.38-86.2 50.2-19.6 130.5-20.9 263.1 30.7l6.6-16.8c-91.7-32.6-187-64.8-276.3-30.7zm260.4 76c-177.5 91.5-260 65.2-352.46 41.4l-4.48 17.5c15.96 4.1 31.94 8.3 48.39 11.9-19.36 13.6-28.82 17.2-40.19 24.6 25.65-1.1 42.18-9.2 59.71-16.6-12.58 15.5-23.55 31-46 47.4 27.02-7.2 53.04-15.2 72.63-28.8-12.47 20-27.72 39.5-44.89 58.6 29.43-9.2 51.69-31.4 74.79-53.4-.3 19.1-9.2 38.2-15.5 57.3 21-19.3 30-33.2 42.7-52.5-1.2 29.2 7 52.8 14.2 82 4.5-27.9 9.5-55.9 9.5-82.9 15 25 35.1 47.8 60.4 68.1-13.5-23.5-27-46.9-34.1-71.3 25.8 24.5 52.7 48.3 85 68.2-21.5-23.5-41.9-47.3-55.1-72.7 23.2 21.8 46.5 43.6 85.3 56.2-28.1-19.3-46.1-41.2-59.3-64.3 26.7 22.4 56.6 42.3 92.7 57.2-31.9-22-49.9-44-62.2-66 26.4 21.8 56.6 36.2 82.9 50.6-20.8-19.2-43.5-37.6-54.4-60.4 21.5 18.7 46.9 26.8 72 35.5-23.3-15.5-36.7-26-50.6-44.3 22.2 11.2 40.7 15.6 67.4 22.2-17.4-9.7-35.9-23-47.8-34.5 26.7 6.7 56 7.8 83.6 7.4-25.3-6.6-52.8-12.4-67.4-22.5 58.4.5 62-4.6 86.5-16.8-51.8 3.8-84.7 2.9-103.3-19.1z"
              />
            </svg>
            Lashes
          </a>
        </div>
        <div>
          <a
            href="/#"
          >
            <svg
              data-testid="subnavigation-icon-2"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="5"
              />
              <line
                x1="12"
                x2="12"
                y1="1"
                y2="3"
              />
              <line
                x1="12"
                x2="12"
                y1="21"
                y2="23"
              />
              <line
                x1="4.22"
                x2="5.64"
                y1="4.22"
                y2="5.64"
              />
              <line
                x1="18.36"
                x2="19.78"
                y1="18.36"
                y2="19.78"
              />
              <line
                x1="1"
                x2="3"
                y1="12"
                y2="12"
              />
              <line
                x1="21"
                x2="23"
                y1="12"
                y2="12"
              />
              <line
                x1="4.22"
                x2="5.64"
                y1="19.78"
                y2="18.36"
              />
              <line
                x1="18.36"
                x2="19.78"
                y1="5.64"
                y2="4.22"
              />
            </svg>
            Tanning
          </a>
        </div>
        <div>
          <a
            href="/#"
          >
            <svg
              data-testid="subnavigation-icon-3"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4 17m-1 0a1 1 0 1 0 2 0a1 1 0 1 0 -2 0"
              />
              <path
                d="M9 5m-1 0a1 1 0 1 0 2 0a1 1 0 1 0 -2 0"
              />
              <path
                d="M4 22l4 -2v-3h12"
              />
              <path
                d="M11 20h9"
              />
              <path
                d="M8 14l3 -2l1 -4c3 1 3 4 3 6"
              />
            </svg>
            Massage
          </a>
        </div>
        <div>
          <a
            href="/#"
          >
            <svg
              data-testid="subnavigation-icon-4"
              fill="currentColor"
              height="24"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 32 32"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M 19.210938 7.0097656 C 18.953605 7.0020895 18.68985 7.0166835 18.421875 7.0566406 C 17.736865 7.1586052 17.093941 7.4137992 16.433594 7.625 C 15.773247 7.8362008 15.119109 8 14.630859 8 C 14.31959 8 13.757226 7.9918469 13.189453 7.9980469 C 12.62168 8.0042469 12.078675 8.0122369 11.599609 8.0996094 C 10.753303 8.2542139 10.020332 8.625504 9.5058594 9.1914062 C 8.9913863 9.7573086 8.7176772 10.467347 8.5917969 11.205078 A 1.0001 1.0001 0 0 0 8.5898438 11.21875 L 7.9882812 15.035156 C 4.8444092 15.212995 2.5666185 17.144707 1.3320312 19.738281 C 0.78950854 20.877379 0.9239394 22.136783 1.5234375 23.058594 C 2.1229356 23.980405 3.1143224 24.604998 4.2539062 24.851562 C 12.426691 26.619913 19.429095 26.079361 27.046875 24.998047 A 1.0001 1.0001 0 0 0 27.0625 24.998047 A 1.0001 1.0001 0 0 0 27.072266 24.998047 L 27.072266 24.996094 A 1.0001 1.0001 0 0 0 27.117188 24.992188 C 27.125187 24.991087 27.132645 24.991417 27.140625 24.990234 A 1.0001 1.0001 0 0 0 27.160156 24.986328 A 1.0001 1.0001 0 0 0 27.443359 24.894531 C 28.735348 24.625296 29.700395 24.082877 30.257812 23.347656 C 30.860807 22.552353 31 21.66 31 20.982422 C 31 17.581633 28.382305 13.481886 23.554688 12.179688 L 22.96875 9.90625 A 1.0001 1.0001 0 0 0 22.962891 9.8867188 C 22.50168 8.2374833 21.012265 7.0634987 19.210938 7.0097656 z M 19.185547 8.9941406 C 20.213061 8.9797344 20.778465 9.5093069 21.035156 10.423828 C 21.035156 10.423828 21.037109 10.425781 21.037109 10.425781 L 21.730469 13.107422 L 21.732422 13.113281 A 1.0001 1.0001 0 0 0 21.798828 13.376953 L 21.962891 14.009766 A 1.0001 1.0001 0 0 0 21.982422 14.091797 C 22.094639 14.493592 21.739122 15.292796 21.267578 15.568359 C 20.980272 15.736026 19.866822 16.159119 18.572266 16.462891 C 17.359672 16.74743 15.939127 16.961012 14.642578 16.986328 C 13.398 16.090951 11.930893 15.358691 10.003906 15.101562 L 10.564453 11.542969 C 10.646573 11.0617 10.791098 10.749707 10.984375 10.537109 C 11.177652 10.324512 11.442291 10.160799 11.958984 10.066406 C 12.126919 10.035776 12.67896 10.003856 13.210938 9.9980469 C 13.742915 9.9922421 14.284129 10 14.630859 10 C 15.51161 10 16.326816 9.7602992 17.042969 9.53125 C 17.759122 9.3022008 18.400807 9.0802385 18.716797 9.0332031 C 18.882911 9.0084342 19.038759 8.9961987 19.185547 8.9941406 z M 24.152344 14.490234 C 27.38392 15.858629 29 18.879672 29 20.982422 C 29 21.392843 28.92003 21.803012 28.664062 22.140625 C 28.411965 22.473135 27.933053 22.819566 26.867188 23 C 23.845001 23.211656 21.914717 22.680867 20.386719 21.792969 C 21.646731 21.526478 22.774012 21.123908 23.521484 20.6875 A 1.0001 1.0001 0 0 0 23.523438 20.685547 C 24.672021 20.01418 25.250057 18.645011 24.892578 17.359375 L 24.152344 14.490234 z M 22.730469 16.982422 L 22.960938 17.876953 A 1.0001 1.0001 0 0 0 22.964844 17.894531 C 23.080218 18.308322 22.914494 18.724696 22.513672 18.958984 C 21.841611 19.351363 19.86932 19.99849 18.138672 20.080078 C 17.957383 19.912952 17.776919 19.74692 17.595703 19.574219 C 17.339396 19.329954 17.061359 19.074354 16.796875 18.822266 C 17.574735 18.717286 18.331995 18.57378 19.029297 18.410156 C 20.44299 18.078428 21.551697 17.717256 22.275391 17.294922 A 1.0001 1.0001 0 0 0 22.277344 17.294922 C 22.436151 17.202117 22.588318 17.097101 22.730469 16.982422 z M 8.4492188 17 C 10.815724 17 12.314065 17.672239 13.615234 18.660156 A 1.0001 1.0001 0 0 0 13.919922 18.892578 C 14.690861 19.51982 15.407825 20.252383 16.214844 21.021484 C 16.465673 21.260528 16.728101 21.499448 17.003906 21.734375 A 1.0001 1.0001 0 0 0 17.326172 21.990234 A 1.0001 1.0001 0 0 0 17.332031 21.994141 C 18.066017 22.589577 18.879367 23.159052 19.8125 23.646484 C 14.902761 24.01751 10.035585 24.058142 4.6757812 22.898438 C 3.9943652 22.751003 3.4632207 22.374689 3.1992188 21.96875 C 2.9352167 21.562811 2.8662414 21.167512 3.1367188 20.599609 C 4.2032721 18.359035 5.8063398 17 8.4492188 17 z"
              />
            </svg>
            Barber
          </a>
        </div>
        <div>
          <a
            href="/#"
          >
            <svg
              data-testid="subnavigation-icon-5"
              fill="currentColor"
              height="24"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15.825.12a.5.5 0 0 1 .132.584c-1.53 3.43-4.743 8.17-7.095 10.64a6.1 6.1 0 0 1-2.373 1.534c-.018.227-.06.538-.16.868-.201.659-.667 1.479-1.708 1.74a8.1 8.1 0 0 1-3.078.132 4 4 0 0 1-.562-.135 1.4 1.4 0 0 1-.466-.247.7.7 0 0 1-.204-.288.62.62 0 0 1 .004-.443c.095-.245.316-.38.461-.452.394-.197.625-.453.867-.826.095-.144.184-.297.287-.472l.117-.198c.151-.255.326-.54.546-.848.528-.739 1.201-.925 1.746-.896q.19.012.348.048c.062-.172.142-.38.238-.608.261-.619.658-1.419 1.187-2.069 2.176-2.67 6.18-6.206 9.117-8.104a.5.5 0 0 1 .596.04M4.705 11.912a1.2 1.2 0 0 0-.419-.1c-.246-.013-.573.05-.879.479-.197.275-.355.532-.5.777l-.105.177c-.106.181-.213.362-.32.528a3.4 3.4 0 0 1-.76.861c.69.112 1.736.111 2.657-.12.559-.139.843-.569.993-1.06a3 3 0 0 0 .126-.75zm1.44.026c.12-.04.277-.1.458-.183a5.1 5.1 0 0 0 1.535-1.1c1.9-1.996 4.412-5.57 6.052-8.631-2.59 1.927-5.566 4.66-7.302 6.792-.442.543-.795 1.243-1.042 1.826-.121.288-.214.54-.275.72v.001l.575.575zm-4.973 3.04.007-.005zm3.582-3.043.002.001h-.002z"
              />
            </svg>
            Beauty
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
